pkg/providers/goinstall.go

```diff
@@ -0,0 +1,126 @@
+package providers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/apex/log"
+)
+
+type goinstall struct {
+	name, repo, tag, latestURL string
+}
+
+func parseRepo(path string) (string, string, string, string) {
+	repo := path
+	tag := "latest"
+	if i := strings.LastIndex(path, "@"); i > -1 {
+		repo = filepath.Clean(path[:i])
+		tag = path[i+1:]
+	}
+
+	name := path
+	if i := strings.LastIndex(repo, "/"); i > -1 {
+		name = repo[i+1:]
+	}
+
+	latestURL := fmt.Sprintf("https://proxy.golang.org/%s/@latest", repo)
+
+	return repo, tag, name, latestURL
+}
+
+func newGoInstall(repo string) (Provider, error) {
+	repoUrl := strings.TrimPrefix(repo, "goinstall://")
+	repo, tag, name, latestURL := parseRepo(repoUrl)
+	return &goinstall{repo: repo, tag: tag, name: name, latestURL: latestURL}, nil
+}
+
+func (g *goinstall) Fetch(opts *FetchOpts) (*File, error) {
+
+	if (len(g.tag) > 0 && g.tag != "latest") || len(opts.Version) > 0 {
+		if len(opts.Version) > 0 {
+			// this is used by for the `ensure` command
+			g.tag = opts.Version
+		}
+		log.Infof("Getting %s release for %s", g.tag, g.repo)
+	} else {
+		log.Infof("Getting latest release for %s", g.repo)
+		if name, _, err := g.GetLatestVersion(); err != nil {
+			return nil, fmt.Errorf("failed to get latest version: %w", err)
+		} else {
+			g.tag = name
+		}
+	}
+
+	cmd := exec.Command("go", "install", fmt.Sprintf("%s@%s", g.repo, g.tag))
+
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+
+	if err := cmd.Run(); err != nil {
+		return nil, fmt.Errorf("failed to install package: %w", err)
+	}
+
+	goBinPath := fmt.Sprintf("%s/bin/%s", os.Getenv("GOPATH"), g.name)
```

Comment:
`GOPATH` could not be set. I think we should do a bit more checks according to the `go install` help: 

> Executables are installed in the directory named by the GOBIN environment
variable, which defaults to $GOPATH/bin or $HOME/go/bin if the GOPATH
environment variable is not set. Executables in $GOROOT
are installed in $GOROOT/bin or $GOTOOLDIR instead of $GOBIN.